pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -210,3 +219,85 @@ def visit_comprehension(self, node: astroid.Comprehension) -> None:
 
                 self.add_message("consider-using-dict-items", node=node)
                 return
+
+    @utils.check_messages("consider-using-str-partition")
+    def visit_subscript(self, node: astroid.Subscript) -> None:
+        """Add message when accessing first or last elements of a str.split()."""
+        assignment = None
+        subscripted_object = node.value
+        if isinstance(subscripted_object, astroid.Name):
+            # Check assignment of this subscripted object
+            assignment_lookup_results = node.value.lookup(node.value.name)
+            if (
+                len(assignment_lookup_results) == 0
+                or len(assignment_lookup_results[-1]) == 0
+            ):
+                return
+            assign_name = assignment_lookup_results[-1][-1]
+            if not isinstance(assign_name, astroid.AssignName):
+                return
+            assignment = assign_name.parent
+            if not isinstance(assignment, astroid.Assign):
+                return
+            # Set subscripted_object to the assignment RHS
+            subscripted_object = assignment.value
```

Comment:
```py
split1 = SEQ.split(',')
get_first = split1[0]
```
If the index lookup happens after it was assigned to a variable, the chance is pretty high that
- it's also used differently, i.e., other elements are needed as well
- or the list is altered before eventually using the index lookup.

IMO this would add either more complexity or false-positives. Both are probably not worth it. What do you think?

Reply:
@cdce8p 
I thought about how I would normally use .split and I agree with both points.

I was also considering to switch the logic instead to visit_call() and check if it's a call to split/rsplit, 
and then check for usage before firing the error. It makes more sense after thinking about it more that the error is fired on the line where .split is used, instead of firing where the subscript is used.

That way the point which you've highlighted (it is used differently / other elems are needed) -- we can account for that and not trigger the error

Would that be a better approach instead?

Reply:
Moving to `visit_call` would indeed make sense. I would suggest to then check if it's part of a subscript and only then continue.

Reply:
> Moving to `visit_call` would indeed make sense. I would suggest to then check if it's part of a subscript and only then continue.

let me give that a shot! :) 

Still not entirely sure how to do so but I'll consult the docs first and then ask for help if I'm really stumped :laughing: 

thanks a lot!